packages/mobile/src/store/offline-downloads/sagas/offlineQueueSagas/workers/downloadTrackWorker.ts

```diff
@@ -47,54 +50,67 @@ function* shouldAbortDownload(trackId: ID) {
   }
 }
 
-export function* downloadTrackWorker(trackId: ID) {
-  const queueItem: OfflineJob = { type: 'track', id: trackId }
+export function* downloadTrackWorker(trackId: ID, requeueCount: number) {
+  const queueItem: OfflineJob = { type: 'track', id: trackId, requeueCount }
   track(
     make({ eventName: EventNames.OFFLINE_MODE_DOWNLOAD_START, ...queueItem })
   )
   yield* put(startJob(queueItem))
 
-  const { jobResult, cancel, abort } = yield* race({
-    jobResult: call(downloadTrackAsync, trackId),
-    abort: call(shouldAbortDownload, trackId),
-    cancel: take(SET_UNREACHABLE)
-  })
-
-  if (abort) {
-    yield* call(removeDownloadedTrack, trackId)
-    yield* put(requestProcessNextJob())
-  } else if (cancel) {
-    yield* put(cancelJob(queueItem))
-    yield* call(removeDownloadedTrack, trackId)
-  } else if (jobResult === OfflineDownloadStatus.ERROR) {
-    track(
-      make({
-        eventName: EventNames.OFFLINE_MODE_DOWNLOAD_FAILURE,
-        ...queueItem
-      })
-    )
-    yield* put(errorJob(queueItem))
-    yield* call(removeDownloadedTrack, trackId)
-    yield* put(requestProcessNextJob())
-  } else if (jobResult === OfflineDownloadStatus.ABANDONED) {
-    track(
-      make({
-        eventName: EventNames.OFFLINE_MODE_DOWNLOAD_FAILURE,
-        ...queueItem
-      })
-    )
-    yield* put(abandonJob(queueItem))
-    yield* call(removeDownloadedTrack, trackId)
-    yield* put(requestProcessNextJob())
-  } else if (jobResult === OfflineDownloadStatus.SUCCESS) {
-    track(
-      make({
-        eventName: EventNames.OFFLINE_MODE_DOWNLOAD_SUCCESS,
-        ...queueItem
-      })
-    )
-    yield* put(completeJob({ ...queueItem, completedAt: Date.now() }))
-    yield* put(requestProcessNextJob())
+  for (let retryCount = 0; retryCount < MAX_RETRY_COUNT; retryCount++) {
```

Comment:
so is it possible to do retries just for the jobResult? i feel like abort and cancel happen when they should, like when we abort, we should not retry. maybe we can huddle about this?

Reply:
Yeah I think you're right. I didn't figure out the structure in the moment, but I can see it now.